Additional clarity (#7)
bkhabs/diem-unified-charter: diem-unified-charter.md

## Background

An emblem is a device, symbol, or figure adopted and used as an identifying mark.
In culture, emblems such as a flag, badge or coat of arms communicate group identity.
In speech, emblems are specific nonverbal gestures or signals that have a direct verbal translation and are widely understood within a particular culture or community.

> The emblems of the International Committee of the Red Cross (ICRC)  are symbols of protection under the Geneva Conventions and are to be worn by all medical and humanitarian personnel and also displayed on their vehicles and buildings while they are in an active warzone, and all military forces operating in an active warzone must not attack entities displaying these emblems.

Author: ```suggestion
The emblems of the International Committee of the Red Cross (ICRC) are symbols of protection under the Geneva Conventions.
```

Author: @mstojens I opted to shorten to eliminate the partial definition, hopefully this would address both of your comments.

> The emblems of the International Committee of the Red Cross (ICRC)  are symbols of protection under the Geneva Conventions and are to be worn by all medical and humanitarian personnel and also displayed on their vehicles and buildings while they are in an active warzone, and all military forces operating in an active warzone must not attack entities displaying these emblems.

Reviewer: ```suggestion

Under International Humanitarian Law (IHL), the red cross, red crescent, and red crystal emblems are symbols of protection.
They can be displayed by medical and humanitarian personnel, units, and transports belonging to States, parties to armed conflict, the International Committee of the Red Cross (ICRC) and other Red Cross and Red Crescent Movement components.
Parties to armed conflict must respect and protect and must not attack entities displaying these emblems.
Similarly, the “blue shield” is another IHL emblem that marks cultural property under special protection.
```

I asked Samit for feedback on these parts and we have some suggestions on phrasing the parts on IHL emblems. One of the most critical aspects of this edit is that it is wrong to say "the emblems of the ICRC," because they're not their emblems. We also thought it would be beneficial to mention another emblem regarded under IHL.

## Introduction

There is a need to sense emblems/symbols through digital communication channels, in contrast to visual or tactile sense and communication.
This need arises due to the increasing relevance and impact of cyber operations on human rights, commerce, and security.
An NFPA 704 Symbol can communicate flammability, health, reactivity and special notices and extensions.
A digital emblem provides a mechanism for communicating the attributes or status of an asset that bears the emblem.
Digital emblems extend the range of identifying marks from the physical realm (visual and tactile) to the digital realm.
The presence of a digital emblem represents a new signal available to cyber operators, one which can be combined with existing operational procedures to reduce uncertainty and improve decision making.

The DIEM working group produces documents which describe the creation, presentation, validation, and interpretation of digital emblems.

## Initial Scope

Under the current charter, the working group is limited to the presentation / discovery of digital emblems for the following digital assets: 

* domain names
* hostnames
* ip addresses

> Designs for QRCodes, NFC, Bluetooth, and other discovery mechanisms are currently out of scope.

Reviewer: ```suggestion
Designs for discovery mechanisms using proximity-based protocols such as QRCodes, NFC, or Bluetooth are currently out of scope.
```

(Reasoning in review comment)

The working group will first describe a high level informational architecture and data model.

> After the architecture and data model are published as an informational RFC, the working group will specify a single concrete serialization and a discovery mechanism based on the DNS.

Reviewer: ```suggestion
After the architecture and data model are published as an informational RFC, the working group will specify a single concrete serialization and discovery mechanisms based on the DNS and optionally other protocols.
```

(Reasoning in review comment)

### Investigation of Serialization Formats

The working group will research, analyze and evaluate application layer serialization formats for suitability to the initial scope.
The working group will attempt to match the conventions for serializing data structures used in the discovery protocols, for example, key value strings in DNS records or structured field values in http.
This work must be completed in the use cases and requirements document, and the architecture must not assume any specific serialization formats or discovery mechanisms.

### Investigation of Securing Mechanisms

The working group will research, analyze and evaluate securing mechanisms for digital emblems.
The group is not required to describe object level securing mechanisms, in the case that existing protocol level securing mechanisms are sufficient, for example DNSSEC for DNS records, or TLS for well known HTTPS URLs.
This work must be completed in the use cases and requirements document, and the architecture must not assume any specific securing mechanism or discovery mechanisms.

### Design of Emblem Types

> The working group will describe at least one digital emblem suitable for indicating protection under international humanitarian law.

Reviewer: ```suggestion
The working group will describe at least one digital emblem suitable for indicating protection under IHL.
```

Now defined as an acronym in my above suggestion. Happy to keep it spelled out, too. Would then not introduce the acronym at all and expand it everywhere.

This document must not assume any specific serialization formats, securing mechanisms, discovery mechanisms, or use cases.
This document must be published as an informational RFC, before any proposed standard documents can be adopted by the working group.

### Initial Serialization

A proposed standard describing the first serialization format for digital emblems.
This document may describe a mandatory to implement securing mechanism.
If a securing mechanism is described, this document must describe at least one mandatory to implement cryptographic algorithm which is already supported by the securing mechanism.
This document must be published as a proposed standard RFC, and must not assume any specific use cases.

### Initial Discovery Mechanism

A proposed standard describing the use of the DNS as the initial presentation/discovery mechanism for digital emblems.
This document may describe a mandatory to implement securing mechanism.
This document must be published as a proposed standard RFC, and must not assume any specific serialization formats, or use cases.

> ### Initial Emblem Types

Author: ```suggestion
### Applicability Statement
```

## Background

An emblem is a device, symbol, or figure adopted and used as an identifying mark.
In culture, emblems such as a flag, badge or coat of arms communicate group identity.
In speech, emblems are specific nonverbal gestures or signals that have a direct verbal translation and are widely understood within a particular culture or community.

> The emblems of the International Committee of the Red Cross (ICRC)  are symbols of protection under the Geneva Conventions and are to be worn by all medical and humanitarian personnel and also displayed on their vehicles and buildings while they are in an active warzone, and all military forces operating in an active warzone must not attack entities displaying these emblems.

Reviewer 2: Probably worth consulting the ICRC participants to review this -- you are correct about "must not attack" but it is also "must actively defend" which I know they're adamant is part of the core definition.

Nit: I would recommend breaking this into 2-3 sentences. Right now, it is pretty long.

> ISO 7010 defines a set of emblems that can be used to identify hazards, these include the skull and crossbones for toxic material, the ionizing radiation symbol and the biological hazard symbol.

Reviewer 2: ```suggestion
ISO 7010 defines a set of emblems that can be used to identify hazards. These include the skull and crossbones for toxic material, the ionizing radiation symbol, and the biological hazard symbol.
```

The white cane is an international emblem for visually impaired pedestrians.
The international symbol for deafness is a white ear with a strip through it on a blue background.
A tactile symbol is used to communicate with vision impaired individuals.

> For example, a trapezoid with bars indicating numbers, or a heart with textures to indicate emotions.

Reviewer 2: I respect the style of one sentence per line, but having the tactile emblem sentences on one line would make this much clearer (I assume the trapezoid has texture also?)

Author: Background context for where I got the trapezoid from: https://www.tsbvi.edu/campus-resources/accessibility/tactile-symbols-library

> A tactile symbol is used to communicate with vision impaired individuals.
> For example, a trapezoid with bars indicating numbers, or a heart with textures to indicate emotions.

Author: ```suggestion
A tactile symbol is used to communicate with vision impaired individuals, for example, a trapezoid with bars indicating numbers, or a heart with textures to indicate emotions.
```

These emblems/symbols require a sense of sight, or touch to become known to the receiver.
"To bear an emblem" means to use or adopt a symbol, figure, or device as an identifying mark. 
To "verify an emblem" means to confirm the authenticity or legitimacy of a particular symbol or design, often by checking its details against a known standard or reference point.

> Emblems may be observed by verifiers without the knowledge of the bearer displaying the emblem, or may be presented to a specific verifier upon request.

Reviewer 2: The group has consensus to use the term "validator" rather than "verifier" so I would recommend use of "validator" throughout.

> Emblems may be observed by verifiers without the knowledge of the bearer displaying the emblem, or may be presented to a specific verifier upon request.

Reviewer 2: ```suggestion
Emblems may be observed by verifiers without the knowledge of the bearer displaying the emblem, or may be presented by the bearer to a specific verifier upon request.
```

Nit: this would be clearer

> To be effective, the semantics of an emblem must be well known, and the emblem must be easily recognizable, and distinguishable from other emblems.

Reviewer 2: ```suggestion
To be effective, the semantics of an emblem must be well known, easily recognizable, and distinguishable from other emblems.
```

Nit

## Introduction

> There is a need to sense emblems/symbols through digital communication channels, in contrast to visual or tactile sense and communication.

Reviewer 2: Is there a reason you are using words other than validate/verify ("sense")? If so, I'm confused about the intent. If not, suggestion to remove these other terms which imply a difference that was not intended.

Same for using symbols in addition to emblems.

Author: I'm trying to follow the dictionary terminology here.
Emblems are symbolic communication, they need a channel to be transmitted to be interacted with.
I'm also trying to highlight that even for physical emblems, we know that they can be interacted with over multiple channels.
The intention here was to leave room for this property to transfer to digital emblems.
How would you change this sentence?

This need arises due to the increasing relevance and impact of cyber operations on human rights, commerce, and security.

> An NFPA 704 Symbol can communicate flammability, health, reactivity and special notices and extensions.

Author: ```suggestion
A single NFPA 704 Symbol can communicate multiple attributes such as flammability, health, reactivity and special notices and extensions.
A a single digital emblem can communicate multiple attributes related to a digital asset.
```

> An NFPA 704 Symbol can communicate flammability, health, reactivity and special notices and extensions.

Reviewer 2: This seems out of place (belongs with the list of emblem examples at the beginning of the doc?)

Author: Thats fair, the reason I put it here, was to highlight that a single emblem can communicate multiple attributes, perhaps I should have just said that.

A digital emblem provides a mechanism for communicating the attributes or status of an asset that bears the emblem.

> Digital emblems extend the range of identifying marks from the physical (visual and tactile) to the digital realm.

Reviewer 2: ```suggestion
Digital emblems extend the range of identifying marks from the physical realm (visual and tactile) to the digital realm.
```

Nit

The presence of a digital emblem represents a new signal available to cyber operators, one which can be combined with existing operational procedures to reduce uncertainty and improve decision making.

> The DIEM working group produces documents which describe the creation, presentation, validation and interpretation of digital emblems.

Reviewer 2: ```suggestion
The DIEM working group produces documents which describe the creation, presentation, validation, and interpretation of digital emblems.
```

Nit

## Initial Scope

Under the current charter, the working group is limited to the presentation / discovery of digital emblems for the following digital assets: 

* domain names

> * hostnames

Reviewer 2: As in machine names, strings usable in DHCP, or limited to the use of the term host within HTTP(S), or something else? This is ambiguous to me, and you clearly meant something that isn't a domain name.

Author: Help me out... where is the consensus on the types of "digital assets" that are in scope?

Reviewer: I was also confused on this bullet point and googled "hostname." It seems to me that "hostname" is not a technical term but more of a concept and as such ambiguous. What would be a hostname that is neither a domain name nor an IP address?

* ip addresses

> Designs for QRCodes, NFC, Bluetooth, and other discovery mechanisms are currently out of scope.

Reviewer 2: ~~I'm confused by the structure of this section: the scope is defined in terms of _assets_ (domain names, hostnames, IP addresses), but the exclusions are then defined in terms of _discovery mechanisms_. If the intent of this statement is to declare "physical assets" out of scope by listing these mechanisms usually associated with those scenarios, I would suggest the following text instead, which also completes the logical set by simply making the negation our of (current) scope without assuming examples.~~

~~Reasoning: who's to say that WG-produced solutions cannot use Bluetooth as a transport even though the asset bearing a digital emblem is an IP address (IP over Bluetooth [is a thing](https://www.rfc-editor.org/rfc/rfc7668))? If an object is hidden from sight and tactile interaction, but the validator establishes Bluetooth communication with it and discovers an emblem for the IP address in use by the Bluetooth peer, that would fit the charter, no?~~

~~```suggestion~~
~~Designs for binding digital emblems to other assets are currently out of scope.~~
~~```~~

Reviewer 2: Sleeping on it, I realize why this is here... given I do not have a better suggestion that won't accomplish the scoping you were intending without re-breaching the definitions of physical/digital assets you deftly dodged, I am simply rescinding this suggestion.

Author: : ) when i wrote this, I had imagined that... after providing emblems for "internet resources" the group might be rechartered to describe how emblems for such resources could be related to emblems that are discovered via proximity channels such as NFC / phone camera's.

In other words, if there is energy to explore this space, the group is forbidden from encroaching on that work, in its initial charter... the door remains open, but a re-charter would be required to walk through it.

The working group will first describe a high level informational architecture and data model.

> After the architecture and data model are published as an information RFC, the working group will specify a single concrete serialization and a discovery mechanism based on the DNS.

Reviewer 3: ```suggestion
After the architecture and data model are published as an informational RFC, the working group will specify a single concrete serialization and a discovery mechanism based on the DNS.
```

The working group will coordinate with relevant working groups within the IETF and other SDOs to ensure that existing standards are leveraged appropriately.
The working group will not produce any standard track generic serialization formats.
The working group will not produce any standard track extensions to the DNS. 
The working group will not develop novel security mechanisms, cryptographic primitives, or digital signature schemes.
Until the architecture document is published, the working group will develop and maintain an informational use cases and requirements document, but is not required to request publication of this document.

### Investigation of Serialization Formats

The working group will research, analyze and evaluate application layer serialization formats for suitability to the initial scope.

> The working group will attempt to match the conventions for serializing data structures used in the discovery protocols, for example, key value strings in DNS records or structured field values in http.

Reviewer 2: This seems to conflict with "will not produce any standard track extensions to the DNS"

Author: See https://github.com/bkhabs/diem-unified-charter/pull/7/files#r1953143650, I welcome suggested text that might make this clearer.

This work must be completed in the use cases and requirements document, and the architecture must not assume any specific securing mechanism or discovery mechanisms.

### Design of Emblem Types

The working group will describe at least one digital emblem suitable for indicating protection under international humanitarian law.
The working group will describe at least one digital emblem suitable for indicating the presence of a hazard or a certification of the absence of a hazard.
This work must be completed in the use cases and requirements document, and must not assume any specific serialization format or discovery mechanisms.

### Use Cases and Requirements

The working group will develop an informational document describing use cases and requirements for the initial proposed standards.
Although this document is listed as a deliverable, it is not required to be published as an RFC.

### Architecture

The working group will develop an informational document describing the architecture, terminology and data model of digital emblems.
This document must not assume any specific serialization formats, securing mechanisms, discovery mechanisms, or use cases.

> This document must be published as an RFC, before any proposed standard documents can be adopted by the working group.

Reviewer 2: ```suggestion
This document must be published as an informational RFC, before any proposed standard documents can be adopted by the working group.
```

Nit (consistency with earlier sentence)

### Initial Serialization

A proposed standard describing the first serialization format for digital emblems.
This document may describe a mandatory to implement securing mechanism.
If a securing mechanism is described, this document must describe at least one mandatory to implement cryptographic algorithm which is already supported by the securing mechanism.

> This document must be published as a proposed standard RFC, and must not assume any specific discovery mechanisms, or use cases.

Author: ```suggestion
This document must be published as a proposed standard RFC, and must not assume any specific use cases.
```

Author: Removing this to enable discovery mechanism and serialization to be more closely aligned, for example "text key value pairs" in _prefix, or "structured field values" in https.

### Initial Discovery Mechanism

> A proposed standard describing the use of the DNS as the initial presentation/discovery mechanism for digital emblems.

Reviewer 2: I'm struggling to align this with the earlier statement about not extending DNS on standards track. Was that statement limited to that section ("not extending in this milestone")?

Also... this seems presumptuous of implementation for non-domain-name assets (IP addresses are in scope). Is it a given the WG will need to start with the DNS to discover emblems associated with IP address assets?

Author: For example making use of RFC8552 is allowed, making updates to RFC8484 or RFC1035 is not.

And yes, to constrain the domain of digital asset to be "internet resources" instead of images, or songs, I added the sentence at the beginning of the initial scope section.

This document may describe a mandatory to implement securing mechanism.
This document must be published as a proposed standard RFC, and must not assume any specific serialization formats, or use cases.

### Initial Emblem Types

> A proposed standard describing the use of the Initial Serialization and Initial Discovery Mechanism for no more than two of the emblem types described in the use cases and requirements document.

Reviewer 2: Why not more than two? That seems arbitrary.

Author: My thought process here, was that we think that specific emblem types require special treatment, and that the group could end up not effectively solving any use cases with any emblem types if it tried to do 100 emblem types in the first charter.

I picked the number 2 arbitrarily, but to allow for the group to address digital alternatives to both the ICRC emblems, and something like the NFPA 704 Symbol.

In other words, a small proposed standard that did not need to do anything other than profile the previous proposed standards for these use cases.

Reviewer 3: Also keep in mind that this is part of the initial scope. When this work is done the group can recharter and do more afterwards.
